File: app/src/entourage/java/social/entourage/android/guide/poi/ReadPoiModule.java
package social.entourage.android.guide.poi;

import dagger.Module;
import dagger.Provides;

/**
 * Module related to ReadPoiFragment
 * @see ReadPoiFragment
 */
@Module
final class ReadPoiModule {
    private final ReadPoiFragment fragment;

    public ReadPoiModule(final ReadPoiFragment fragment) {
        this.fragment = fragment;
    }

    @Provides
    public ReadPoiPresenter providesMainPresenter() {
        return new ReadPoiPresenter(fragment);
    }
}

File: app/src/main/java/social/entourage/android/guide/poi/ReadPoiModule.kt
package social.entourage.android.guide.poi

import dagger.Module
import dagger.Provides
import social.entourage.android.api.request.PoiRequest

/**
 * Module related to ReadPoiFragment
 * @see ReadPoiFragment
 */
@Module
internal class ReadPoiModule(private val fragment: ReadPoiFragment,private val request: PoiRequest) {
    @Provides
    fun providesMainPresenter(): ReadPoiPresenter {
        return ReadPoiPresenter(fragment,request)
    }

}